### DISCUSSION TITLE
Cookie expire at Session instead of date.

### DISCUSSION THREAD

**Discussion by BobSquarePants** [pallets/flask#5042] [Q&A] [answered]
Hi everyone, I create a cookie this way in flask

```python
set_cookie('aName','aValue',expires=unixepoch,samesite="Strict",secure=True,httponly=True)
```
`unixepoch` is either a `int `or a `str` with a unixepoch in the future but still in the browser dev tool the cookie will expire after the session.. any ideas ?

Thanks.

**Comment by davidism** [comment]
I can't reproduce the issue as described. Setting a cookie with an expires date in the future does persist it across browser sessions. Be sure to include a reproducible example when asking a question.

**Comment by BobSquarePants** [comment]
Thanks @davidism 

Here a more complete example
```python
@app.route('/anEndpoint', methods=['GET', 'POST'])
def anEndpoint():
    resp = make_response(redirect('/'))	
    resp.set_cookie('aName','aValue',expires=unixepoch,samesite="Strict",secure=True,httponly=True)
    return resp
```

**Reply by BobSquarePants** [reply]
btw I'm running Flask 2.2.2

**Reply by davidism** [reply]
I can't reproduce the issue with this example. When I use a date in the future for the `unixepoch` variable which you didn't provide, it works as expected.

**Comment by BobSquarePants** [comment, marked as answer]
That Internet browser had in the options "Clear cookies etc.. when closing.." so the browser was setting all cookies and any other storage as "session expire"
